Source/x11/XGServerEvent.m

```diff
@@ -1045,7 +1046,12 @@ - (void) processEvent: (XEvent *) event
 		 * the programa can move/resize the window while we send
 		 * this event, causing a confusion.
 		 */
-		[[NSApp windowWithWindowNumber: cWin->number] sendEvent: r];
+                window = [NSApp windowWithWindowNumber: cWin->number];
+		[window sendEvent: r];
+                /* Update monitor_id of the backend window.
+                   NSWindow may change screen pointer while processing
+                   the event. */
+                cWin->monitor_id = [[window screen] screenNumber];
```

Comment:
Computing the screen of a window may be a rather expensive operation. Do we really want to execute this after every event?

Reply:
Hm... I need to think about it.